src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -533,43 +539,44 @@ public void BlittableValidation(int size)
             int currentSize)
         {
             var offsetCounter = 0;
-            for (var i = numberOfProps; i > 0; i--)
+            for (var i = numberOfProps - 1; i >= 0; i--)
             {
                 int stringLength;
                 var nameOffset = 0;
-                nameOffset = ReadNumber((_mem + propsOffsetList + i * propsNamesOffsetSize),
+                nameOffset = ReadNumber((_mem + propsOffsetList + 1 + i * propsNamesOffsetSize),
                     propsNamesOffsetSize);
                 if ((nameOffset > currentSize) || (nameOffset < 0))
                     throw new InvalidDataException("Properties names offset not valid");
                 stringLength = StringValidation(propsOffsetList - nameOffset);
-                if (offsetCounter + stringLength + 1 != nameOffset)
+                if (offsetCounter + stringLength != nameOffset)
                     throw new InvalidDataException("Properties names offset not valid");
                 offsetCounter = nameOffset;
-                currentSize -= (stringLength + 1);
+                currentSize -= (stringLength);
             }
             return currentSize;
         }
 
         private int StringValidation(int stringOffset)
         {
-            byte offset;
+            byte lenOffset;
+            byte escOffset;
             int stringLength;
-            stringLength = ReadVariableSizeInt(stringOffset, out offset);
+            stringLength = ReadVariableSizeInt(stringOffset, out lenOffset);
             if (stringLength < 0)
                 throw new InvalidDataException("String not valid");
-            var str = stringOffset + offset;
-            var stringEnd = ReadVariableSizeInt(stringOffset + offset + stringLength, out offset);
-            if (stringEnd != 0)
+            var str = stringOffset + lenOffset;
+            var escCount = ReadVariableSizeInt(stringOffset + lenOffset + stringLength, out escOffset);
+            if (escCount != 0)
             {
-                for (var i = 0; i < stringEnd; i++)
+                for (var i = 0; i < escCount; i++)
                 {
-                    var escCharOffset = ReadNumber(_mem + str + stringLength + offset + i, 1);
+                    var escCharOffset = ReadNumber(_mem + str + stringLength + escOffset + i, 1);
                     var escChar = (char)ReadNumber(_mem + str + escCharOffset, 1);
                     if (!(EscapeChars.Contains(escChar)))
```

Comment:
Probably better to do this as a `switch` statement on the raw byte. Will be more effcient.